src/main/java/apoc/algo/algorithms/BetweennessCentrality.java

```diff
@@ -0,0 +1,268 @@
+                algorithm.sourceChunkStartingIndex,
+                algorithm.relationshipTarget);
+    }
+
+    public void computeUnweightedParallel(int [] sourceDegreeData,
+                                  int [] sourceChunkStartingIndex,
+                                  int [] relationshipTarget) {
+        betweennessCentrality = new double[nodeCount];
+        Arrays.fill(betweennessCentrality, 0);
+        long before = System.currentTimeMillis();
+
+        int numOfThreads = Pools.getNoThreadsInDefaultPool();
+        assert(numOfThreads != 0);
+        int batchSize = (int)nodeCount/numOfThreads;
+        int batches = 0;
+        if (batchSize > 0)
+            batches = (int)nodeCount/batchSize;
+
+        if (batchSize < MINIMUM_BATCH_SIZE) {
+            batches = 1;
+            batchSize = nodeCount;
+        }
+
+
+        List<Future> futures = new ArrayList<>(batches);
+        intermediateBcPerThread = new HashMap<>();
+        int nodeIter = 0;
+        int batchNumber = 0;
+        while(nodeIter < nodeCount) {
+            final int start = nodeIter;
+            final int end = Integer.min(start + batchSize, nodeCount);
+            final int threadBatchNo = batchNumber;
+            Future future = pool.submit(new Runnable() {
+                @Override
+                public void run() {
+                    processNodesInBatch(threadBatchNo, start, end, sourceDegreeData, sourceChunkStartingIndex, relationshipTarget);
+                }
+            });
+            nodeIter = end;
+            batchNumber++;
+            futures.add(future);
+        }
+        log.info("Total batches: " + batchNumber);
+        AlgoUtils.waitForTasks(futures);
+
+        compileResults(batchNumber);
+
+        long after = System.currentTimeMillis();
+        long difference = after - before;
+        log.info("Computations took " + difference + " milliseconds");
+        stats.computeMillis = difference;
+    }
+
+    private void compileResults(int batchNumber) {
+        for (int i = 0; i < nodeCount; i++) {
+            double value = 0;
+            for (int batch = 0; batch < batchNumber; batch++) {
+                value += intermediateBcPerThread.get(batch).getOrDefault(i, 0.0);
+            }
+            betweennessCentrality[i] = value;
+        }
+    }
+
+    private void processNodesInBatch(int threadBatchNo,
+                                     int start,
+                                     int end,
+                                     int [] sourceDegreeData,
+                                     int [] sourceChunkStartingIndex,
+                                     int [] relationshipTarget) {
+        Stack<Integer> stack = new Stack<>(); // S
+        Queue<Integer> queue = new LinkedList<>();
+
+        log.info("Thread: " + Thread.currentThread().getName() + " processing " + start + " " + end);
+        Map<Integer, ArrayList<Integer>>predecessors = new HashMap<Integer, ArrayList<Integer>>(); // Pw
```

Comment:
Oh! I was looking for something like this. This would improve memory a lot.
I didn't know neo4j code had this implementation.